contrib/epee/include/string_tools.h

```diff
@@ -353,7 +353,12 @@ POP_WARNINGS
 	//----------------------------------------------------------------------------
 	inline bool validate_hex(uint64_t length, const std::string& str)
 	{
-		return std::regex_match(str, std::regex("'^[0-9abcdefABCDEF]+$'")) && str.size() == length;
+		if (str.size() != length)
```

Comment:
A cpp already exists for this file, might as well move this code to that at the same time.